core/src/main/java/com/linecorp/armeria/internal/common/AbstractHttp2ConnectionHandler.java

```diff
@@ -162,8 +170,16 @@ public final ChannelFuture goAway(ChannelHandlerContext ctx, int lastStreamId, l
         return super.goAway(ctx, lastStreamId, errorCode, debugData, promise);
     }
 
+    protected final void setGoAwayDebugMessage(String debugMessage) {
+        goAwayDebugData = Unpooled.copiedBuffer(debugMessage.getBytes()).asReadOnly();
```

Comment:
- How about using `Unpooled.copiedBuffer(String, Charset)` instead?
- It's probably OK to omit `.asReadOnly()`.
- How about making `goAwayDebugData` unreleasable with `Unpooled.unreleasableBuffer()`, so we don't have to worry about memory leaks?

Reply:
Done. 
Quick question: once we mark it unreleasable is it going to be GC'ed once it's no longer referenced by goAwayDebugData (e.g. after next setGoAwayDebugMessage call replaces it with a new buffer)?

Reply:
Yeah, unpooled buffers will be released by GC. Pooled buffers won't.